ipatests/test_xmlrpc/test_pwpolicy_plugin.py

```diff
@@ -157,6 +157,36 @@ def test_9_pwpolicy_mod(self):
         entry = api.Command['pwpolicy_mod'](self.group, krbminpwdlife=50)['result']
         assert_attr_equal(entry, 'krbminpwdlife', '50')
 
+    def test_maxlife_pwpolicy(self):
+        """Check maxlife error message where minlife > maxlife specified
+
+        When minlife > maxlife specified on commandline, it says:
+        "ipa: ERROR: invalid 'maxlife': Maximum password life must be
+        greater than minimum."
+
+        But when minlife == maxlife specfied, It works.
+        This test check that error message says what exactly it does.
+
+        related: https://pagure.io/freeipa/issue/9038
+        """
+        # when minlife(specified in hours) == maxlife(specified in days)
+        entry = api.Command['pwpolicy_mod'](krbminpwdlife=24,
+                                            krbmaxpwdlife=1)['result']
+        assert_attr_equal(entry, 'krbminpwdlife', '24')
+
+        # when minlife(specified in hours) < maxlife(specified in days)
+        entry = api.Command['pwpolicy_mod'](krbminpwdlife=20,
+                                            krbmaxpwdlife=1)['result']
+        assert_attr_equal(entry, 'krbminpwdlife', '20')
+
+        # when minlife(specified in hours) > maxlife(specified in days)
+        try:
+            entry = api.Command['pwpolicy_mod'](krbminpwdlife=25,
+                                                krbmaxpwdlife=1)['result']
+        except errors.ValidationError as e:
+            assert ("Maximum password life must be equal to "
+                    "or greater than the minimum.") in str(e)
```

Comment:
Is it an expected mandatory exception?
https://docs.pytest.org/en/latest/how-to/assert.html#assertions-about-expected-exceptions

Reply:
Yes, see https://github.com/freeipa/freeipa/pull/6086/files#diff-fcbdf956172aafe4f6c1e64b54988af95c6bd35770c9a1e83e056e72e3114d21R492

Reply:
let me rephrase myself.
if this is an expected exception then it should be raised *always*. Currently, the test assumes that the exception is *optional*. For example, `try except else` construction or Pytest's `pytest.raises` (the link above).

Reply:
Missed your point :)
Updated.

Reply:
example: https://github.com/freeipa/freeipa/blob/637412acfd20eaa9e741db7a5d548d6daa14aebd/ipatests/test_xmlrpc/test_plugins_plugin.py#L56-L60